### PR TITLE
[WIP] 2041: added addFile and updateFile methods with new unit tests

### LINKED ISSUE #2041: Create Submission function for creating / updating the status of a file
## Task

The `Submission` needs function(s) for adding a new `File` to the submission and updating its status as it is uploading, storing and complete. 

### Specification

- The create function should return a file ID to be used by the use-case's to update the state in future.
- The update status function should take the file ID as a parameter

### PR DESCRIPTION
**Pull request by erezmus**
Closes #2041

### CODE REVIEW

File: packages/component-submission/server/aggregates/Submission.js [hunk 1]
```diff
@@ -164,6 +169,42 @@ class Submission {
...
+
+  async updateFile(id, attrs) {
+    const file = find(this.files, { id })
+
+    if (!file) {
+      throw new Error(`Could not find file with id: ${id}`)
+    }
+
+    // manuscriptId cannot be changed, but may that should be done at file model level?
+    Object.assign(file, omit(attrs, 'manuscriptId'))
```
Comment (erezmus) [inline packages/component-submission/server/aggregates/Submission.js:195]: We only ever change the status, so change this to use 'only'

**Comment by erezmus** [comment]
We might need to add internal factory methods for building the models so that if we swap out objection, it will be easier

**Comment by erezmus** [comment]
i'm not sure if this is still needed in light of DDD proposal?

**Comment by diversemix** [comment]
Awaiting outcome of DDD discussion.

/cc @hdrury1

**Comment by erezmus** [comment]
Closing this off for now at it might not relevant any more. Will reopen if needed

**Pull request closed without merging**